### PR TITLE
GALAXY-1842_Web_Articles

### PR DESCRIPTION
**Pull request by TomMarshBBC**

### CODE REVIEW

File: ribbon/src/main/java/bbc/co/uk/nautilusapp/ribbon/ui/RibbonActivity.kt [hunk 1]
```diff
@@ -41,6 +42,11 @@ class RibbonActivity : AppCompatActivity() {
         viewModel.ribbonConfiguration.observe(this, Observer(::onRibbonConfiguration))
     }
 
+    override fun onStart() {
+        super.onStart()
+        warmupChrome()
```
Comment (johnryM) [inline ribbon/src/main/java/bbc/co/uk/nautilusapp/ribbon/ui/RibbonActivity.kt]: 🔥

File: web-browser/src/main/java/bbc/co/uk/webbrowser/WebBrowser.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import androidx.core.content.ContextCompat
+import androidx.core.net.toUri
+import bbc.co.uk.webbrowser.ui.WebBrowserActivity
+import bbc.co.uk.webbrowser.util.UrlHelper.openEmailClient
+import bbc.co.uk.webbrowser.util.UrlHelper.parseLiveUrl
+import bbc.co.uk.webbrowser.util.isEmailLink
+import bbc.co.uk.webbrowser.util.isLiveLink
+import javax.inject.Inject
+
+class WebBrowser
```
Comment (johnryM) [inline web-browser/src/main/java/bbc/co/uk/webbrowser/WebBrowser.kt]: Could this class just be a Kotlin file with some helper functions in it? I'm not sure why it needs to be passed into the Navigator.

File: web-browser/src/main/java/bbc/co/uk/webbrowser/WebBrowserClient.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import bbc.co.uk.webbrowser.ui.WebBrowserActivity
+import kotlinx.android.synthetic.main.acitivity_fallback_web_browser.*
+import javax.inject.Inject
+
+class WebBrowserClient
+@Inject internal constructor(private val activity: WebBrowserActivity) : WebViewClient() {
+
+    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
+        return when {
+            request.url.host.contains(activity.getString(R.string.bbc_local_domain)) -> false
```
[the remark this answers is not in the document]
Comment (TomMarshBBC) [reply web-browser/src/main/java/bbc/co/uk/webbrowser/WebBrowserClient.kt]: if the host was the focus, then I couldn't do `contains` as  part of the statement

File: web-browser/src/main/java/bbc/co/uk/webbrowser/WebBrowser.kt (hunk 2 above)
Comment (TomMarshBBC) [reply web-browser/src/main/java/bbc/co/uk/webbrowser/WebBrowser.kt]: I could do that I guess, but I feel like we should be unit testing the Navigator (eventually) , and this makes it easier to mock dependencies.

**Pull request merged**